syntax = "proto3";

package argussight;

service SpawnerService {
    rpc StartProcesses (StartProcessesRequest) returns (StartProcessesResponse);
    rpc TerminateProcesses (TerminateProcessesRequest) returns (TerminateProcessesResponse);
    rpc ManageProcesses (ManageProcessesRequest) returns (ManageProcessesResponse);
}

message StartProcessesRequest {}
message StartProcessesResponse {
    string message = 1;
}

message TerminateProcessesRequest {
    repeated string names = 1;
}
message TerminateProcessesResponse {
    string message = 1;
}

message ManageProcessesRequest {}
message ManageProcessesResponse {
    string message = 1;
}